### DISCUSSION TITLE
Custom device properties

### DISCUSSION THREAD

**Discussion by RobertJoonas** [plausible/analytics#2466] [Anything Else]
I understand you are primarily focused on web analytics, but I have seen in your documentation that you target both websites and also apps. My use-case is basically no different from an app. If I'd create a mobile app, I'd also want to submit my own data about the device the user is using, like if they're using a Snapdragon or Mediatek chip, or other device data that helps to further optimize the app and see how users with different hardware use the app differently.

It seems unnecessarily inflexible that the "Device" data is specifically limited to "Browser", "OS" and "Screen Size". I think you could easily make it possible to submit custom properties there, because it basically already is supported through the "screen_width" parameter, right? The screen width is not read from the user agent, instead there is a separate JSON parameter that can be used with the Events API to submit the screen width, and that screen width is then categorized and shown in the "Devices" area of the plausible dashboard. Would it be any technical challenge to just allow submitting an arbitraty number of custom parameters that work exactly the same like the screen_width parameter?

To give an example, this is the Events API example from your documentation:

curl -i -X POST https://plausible.io/api/event \ -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.121 Safari/537.36 OPR/71.0.3770.284' \ -H 'X-Forwarded-For: 127.0.0.1' \ -H 'Content-Type: application/json' \ --data '{"name":"pageview","url":["http://dummy.site"](http://dummy.site/),"domain":"dummy.site","screen_width":1666}'

So that screen_width is a separate parameter. Would there be any problem with simply making something like this possible?

curl -i -X POST https://plausible.io/api/event \ -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.121 Safari/537.36 OPR/71.0.3770.284' \ -H 'X-Forwarded-For: 127.0.0.1' \ -H 'Content-Type: application/json' \ --data '{"name":"pageview","url":["http://dummy.site"](http://dummy.site/),"domain":"dummy.site","screen_width":1666,"Input Device":"Gamepad"}'

That would seem like a really simple and elegant solution to me that should probably not need more than a few lines of code on your side to implement, because it's just additional parameters that work exactly the same like the screen_width parameter internally, and you have all the support for that in your backend and dashboard already. It just additionally needs to work with custom parameters instead of only the hardcoded screen_width one

(feature request via email)

**Comment by RobertJoonas** [comment]
Sharing an idea from the customer on how this could look on the dashboard:

![image](https://user-images.githubusercontent.com/56999674/203812121-450654c7-9398-4cf0-a337-8bc44e4e9748.png)

The additional `CPU Type` and `Input Device` properties would only show up if there are such properties in the stats.

Alternatively, all the custom device properties could be listed under a dropdown called "Custom" which would only show up if there are custom properties in the stats.

**Comment by luxaritas** [comment]
Additionally, see #1025 Device type is now coming from user-agent, and even if it wasn't it would be helpful to have granular device size info here (which I'll be adding to my custom props).
